partiql-app/partiql-cli/src/main/kotlin/org/partiql/Main.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ * A copy of the License is located at:
+ *
+ *      http://aws.amazon.com/apache2.0/
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+ *  language governing permissions and limitations under the License.
+ */
+@file:JvmName("Main")
+
+@file:Suppress("DEPRECATION")
+
+package org.partiql
+
+import com.amazon.ion.system.IonSystemBuilder
+import org.partiql.lang.eval.ExprValueFactory
+import org.partiql.pico.CommandPartiQL
+import org.partiql.pico.CommandQuery
+import org.partiql.pico.CommandShell
+import picocli.CommandLine
+import kotlin.system.exitProcess
+
+/**
+ * Runs the PartiQL CLI.
+ */
+fun main(args: Array<String>) {
+    val ion = IonSystemBuilder.standard().build()
+    val valueFactory = ExprValueFactory.standard(ion)
+    val command = CommandLine(CommandPartiQL())
+        .addSubcommand(CommandQuery(valueFactory))
+        .addSubcommand(CommandShell(valueFactory))
```

Comment:
Rather than subcommands, what do you think about running the shell if no query is provided? You can still provide the environment, but if no input file or query text is provided, then you get the REPL. This is typically how REPLs work such as python, ruby, lua, etc. I suppose ours is a _bit_ odd because the query is provided as a text parameter rather than a file path .. something we remedy by using stdin 

So the usage would be
```shell
# executes `select * from foo`
echo "select * from foo" | partiql

# executes the query within `foo.partiql`
partiql ./foo.partiql

# executes the repl
partiql
```

This is consistent with python, ruby, lua, and I presume many other interpreters with `(print (eval (read)))`..

Reply:
Ack. I like this idea.

I've updated the CLI to receive an optional file (arity "0..1") as a parameter. Performed the changes so we can read queries from standard input. Now, if the non-interactive configs haven't been modified, a file isn't passed as a parameter, and nothing is in standard input, we start the shell.